src/OptionT.ts

```diff
@@ -1,21 +1,39 @@
-import { getSome } from './option/some';
-import { getNone } from './option/none';
 
-/**
- * An interface describing the argument passed to [[OptT]]'s `match` function.
- */
 export interface OptMatch<T, U, V> {
   some: (val: T) => U;
   none: () => V;
 }
 
+
 /**
- * An interface describing the `OptionT` type.
- *
- * Items of this type can either be a `Some` value (with something of
- * type `T` in them), or a `None` value (with nothing in them).
+ * An interface describing the argument passed to [[OptT]]'s `match` function.
  */
-export interface OptT<T> {
+
+export default abstract class OptionT<T> {
+  constructor(value?: T) {
+  }
+
+  static of<T>(value?: T): OptionT<T> {
+    if (value === null || typeof value === 'undefined') {
+      return new None();
+    }
+    return new Some(value);
+  }
+
+  static some<T>(value: T): OptionT<T> {
+    if (value === null || typeof value === 'undefined') {
+      throw('Cannot create a Some of a null or undefined value');
+    }
+    return new Some(value);
+  }
+
+  static none<T>(value?: T): OptionT<T> {
+    if (value === null || typeof value === 'undefined') {
+      return new None();
+    }
+    throw('Cannot create a None of a non-null or undefined value');
+  }
+
   /**
    * Returns `true` if this [[OptT]] is a `Some`, returns false if it is a `None`.
```

Comment:
Do you mind doing a find-all + replace-all on these [[OptT]] bits?
Should be valid to replace all occurrences of `[[OptT]]` with `[[OptionT]]`.